### PR TITLE
Add node-like GOSH executor integration tests

### PR DESCRIPTION
**Pull request by awnion**
## Summary
- Add integration-style executor coverage under `tvm_executor/tests/` for node-like GOSH execution paths.
- Introduce shared test fixtures for `ExecuteParams`, active accounts, inbound messages, block observations, and checked-in WASM cache setup.
- Wire `tvm_executor` to forward the `gosh` feature to `tvm_vm` and add test-only ABI/hash dependencies.
- Remove the inline executor-boundary WIP from `transaction_executor.rs`.

## Testing
- `cargo test -p tvm_executor --features gosh,wasmtime --test node_executor_gosh`
- `cargo test -p tvm_executor --features gosh,wasmtime`
- `cargo test -p tvm_executor`

### CODE REVIEW

**Comment by awnion** [comment]
## Code review

The integration-test skeleton is solid: it exercises `OrdinaryTransactionExecutor::execute_with_libs_and_params(Some(msg), ...)` through real accounts, inbound messages, VM execution, action lists, account updates, outbound messages, and the WASM path. A few details still make the harness less node-like than the test names imply.

### Findings

1. `available_credit` is synthetic in one no-dapp path.

   `gosh_action_failure_rolls_back_observable_effects` sets `available_credit = 13` while `dapp_id` is absent. The node uses `0` when there is no `dapp_id`; it only reads dapp-config credit when a dapp id is present. I left an applyable suggestion on the exact line.

   The `13` in the dapp test is fine as a synthetic cap-path fixture, but it should be documented as such or produced by a small helper that models “dapp config returned 13”.

2. Forward prices are all-zero.

   `executor_config()` uses `MsgForwardPrices::default()` for config params 24/25. That bypasses realistic forwarding-fee behavior. The node uses the SDK’s internal `default_mc()` / `default_wc()` values, but that trait is `pub(crate)`, so integration tests cannot import it directly.

   Suggested helper snippet:

   ```rust
   fn default_mc_msg_forward_prices() -> MsgForwardPrices {
       MsgForwardPrices {
           lump_price: 10_000_000,
           bit_price: 655_360_000,
           cell_price: 65_536_000_000,
           ihr_price_factor: 98_304,
           first_frac: 21_845,
           next_frac: 21_845,
       }
   }

   fn default_wc_msg_forward_prices() -> MsgForwardPrices {
       MsgForwardPrices {
           lump_price: 1_000_000,
           bit_price: 65_536_000,
           cell_price: 6_553_600_000,
           ihr_price_factor: 98_304,
           first_frac: 21_845,
           next_frac: 21_845,
       }
   }
   ```

   I left an applyable suggestion for using these helpers in `executor_config()`.

3. Outbound helper forces masterchain destinations.

   `outbound_internal_message()` always uses `masterchain_address(dst)`. That avoids missing workchain config, but it also drops workchain-0 outbound coverage. Better to make the workchain explicit and cover wc0 in at least one non-WASM scenario.

   Suggested helper shape:

   ```rust
   pub fn outbound_internal_message(dst: u8, value: u64, workchain_id: i8) -> Message {
       let mut header = InternalMessageHeader::new();
       header.ihr_disabled = true;
       header.set_dst(MsgAddressInt::with_standart(
           None,
           workchain_id as i32,
           UInt256::with_array([dst; 32]).into(),
       ).unwrap());
       header.value = CurrencyCollection::with_grams(value);
       Message::with_int_header(header)
   }
   ```

   I left an applyable suggestion for the helper. Follow-up needed: add `send_action_wc(...)` and use wc0 in at least one test.

4. `RUNWASM` stack layout needs a comment.

   The current order is correct but easy to break. I left an applyable suggestion documenting the top-to-bottom mapping before `RUNWASM`.

5. Trace-callback executor coverage was lost.

   The deleted inline test exercised `debug = true` / `trace_callback = Some(...)` through `execute_with_libs_and_params`. Add an integration test that sets a callback, runs a small contract, and asserts the callback count is non-zero.

   Suggested test snippet:

   ```rust
   #[test]
   fn trace_callback_runs_through_full_executor() {
       let harness = ExecutionHarness::default();
       let trace_steps = Arc::new(AtomicU64::new(0));
       let callback_steps = trace_steps.clone();
       let trace_callback: Arc<tvm_vm::executor::TraceCallback> = Arc::new(move |_, _| {
           callback_steps.fetch_add(1, Ordering::Relaxed);
       });

       let mut params = NodeExecuteParamsFixture::default();
       params.trace_callback = Some(trace_callback);
       params.last_tr_lt = Arc::new(AtomicU64::new(1_400));

       let code = compile_code("PUSHINT 1\n");
       let observation = harness.run(code, internal_message(1, 7, 1_000_000_000), params).unwrap();

       assert!(!observation.ordinary_description().aborted);
       assert!(trace_steps.load(Ordering::Relaxed) > 0);
   }
   ```

6. WASM fixtures are eagerly precompiled for non-WASM tests.

   `NodeExecuteParamsFixture::default()` builds a Wasmtime engine and precompiles components whenever `wasmtime` is enabled, even for the non-WASM tests. This is not a correctness bug, but it is unnecessary and unlike the node’s shared cache.

   Suggested direction: make WASM fixtures optional or share them through `OnceLock` so non-WASM tests do not pay the setup cost.

7. `read_state_update()` should be strict.

   `MockBlockSink::consume` silently substitutes zero hashes if `read_state_update()` fails. `execute_with_libs_and_params` must write a state update, so this should fail loudly. I left an applyable suggestion on the exact lines.

### Non-blocking follow-ups

I would not block this PR on node-side `minted_shell -= gas_used`, `prev_trans_hash/lt`, external-message coverage, or the full node WASM mandatory-hash set. Those are useful follow-ups if the goal expands from executor-boundary coverage to block-builder fidelity, but they are not required for this integration-test skeleton.

File: tvm_executor/tests/node_executor_gosh.rs [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+fn gosh_action_failure_rolls_back_observable_effects() {
+    let harness = ExecutionHarness::default();
+    let last_tr_lt = Arc::new(AtomicU64::new(600));
+    let vm_execution_is_block_related = Arc::new(Mutex::new(false));
+
+    let mut params = NodeExecuteParamsFixture::default();
+    params.block_unixtime = 400;
+    params.last_tr_lt = last_tr_lt;
+    params.vm_execution_is_block_related = vm_execution_is_block_related;
+    params.available_credit = 13;
```
Comment (awnion) [inline tvm_executor/tests/node_executor_gosh.rs]: This test has no `dapp_id`, so the node would pass `available_credit = 0`. Keeping `13` here exercises a synthetic path rather than the node-like path.

```suggestion
    // Node uses available_credit = 0 when dapp_id is absent.
    params.available_credit = 0;
```

**Review by awnion — COMMENTED** [review]
Line-level suggestions for the refined review items.

File: tvm_executor/tests/common/mod.rs [hunk 2]
```diff
@@ -0,0 +1,529 @@
...
+        vm_execution_is_block_related: Arc<Mutex<bool>>,
+        block_collation_was_finished: Arc<Mutex<bool>>,
+        old_account_hash: UInt256,
+        original_balance: CurrencyCollection,
+    ) -> ExecutionObservation {
+        let out_messages = collect_out_messages(&transaction);
+        let state_update = transaction.read_state_update().unwrap_or_else(|_| HashUpdate {
+            old_hash: UInt256::default(),
+            new_hash: UInt256::default(),
+        });
```
Comment (awnion) [inline tvm_executor/tests/common/mod.rs]: `execute_with_libs_and_params` must write a state update, so the harness should fail loudly if it is missing. The zero-hash fallback would make a real executor regression harder to diagnose.

```suggestion
        let state_update = transaction
            .read_state_update()
            .expect("executor must write state_update");
```

File: tvm_executor/tests/common/mod.rs [hunk 3]
```diff
@@ -0,0 +1,529 @@
...
+    config.set_config(ConfigParamEnum::ConfigParam20(gas_prices())).unwrap();
+    config.set_config(ConfigParamEnum::ConfigParam21(gas_prices())).unwrap();
+    config.set_config(ConfigParamEnum::ConfigParam24(MsgForwardPrices::default())).unwrap();
+    config.set_config(ConfigParamEnum::ConfigParam25(MsgForwardPrices::default())).unwrap();
+    config.set_config(ConfigParamEnum::ConfigParam31(ConfigParam31::new())).unwrap();
+    BlockchainConfig::with_config(config).unwrap()
+}
+
+#[cfg(all(feature = "gosh", feature = "wasmtime"))]
+pub fn wasm_validation_code(call: &WasmCallFixture, actions: OutActions) -> String {
```
Comment (awnion) [inline tvm_executor/tests/common/mod.rs:542]: The `RUNWASM` stack ordering is correct but non-obvious. Please document the top-to-bottom mapping so this does not get broken by a future fixture edit.

```suggestion
// Stack top-to-bottom before RUNWASM:
//   var(0) = empty executable cell, so RUNWASM uses hash mode
//   var(1) = wasm interface / instance name
//   var(2) = wasm function name
//   var(3) = ABI-encoded wasm args
//   var(4) = ABI-encoded wasm hash
// RUNWASM fetches five stack items and reads them by those var indexes.
pub fn wasm_validation_code(call: &WasmCallFixture, actions: OutActions) -> String {
```

File: tvm_executor/tests/common/mod.rs [hunk 4]
```diff
@@ -0,0 +1,529 @@
...
+    config
+        .set_config(ConfigParamEnum::ConfigParam8(ConfigParam8 {
+            global_version: GlobalVersion { version: 42, capabilities: 0x572e },
+        }))
+        .unwrap();
+    config.set_config(ConfigParamEnum::ConfigParam18(storage_prices())).unwrap();
+    config.set_config(ConfigParamEnum::ConfigParam20(gas_prices())).unwrap();
+    config.set_config(ConfigParamEnum::ConfigParam21(gas_prices())).unwrap();
+    config.set_config(ConfigParamEnum::ConfigParam24(MsgForwardPrices::default())).unwrap();
+    config.set_config(ConfigParamEnum::ConfigParam25(MsgForwardPrices::default())).unwrap();
```
Comment (awnion) [inline tvm_executor/tests/common/mod.rs]: These defaults are all-zero. Use node-like forwarding prices here; add the two helper functions from the top-level review comment nearby.

```suggestion
    config.set_config(ConfigParamEnum::ConfigParam24(default_mc_msg_forward_prices())).unwrap();
    config.set_config(ConfigParamEnum::ConfigParam25(default_wc_msg_forward_prices())).unwrap();
```

File: tvm_executor/tests/common/mod.rs [hunk 5]
```diff
@@ -0,0 +1,529 @@
...
+    ))
+}
+
+pub fn outbound_internal_message(dst: u8, value: u64) -> Message {
+    let mut header = InternalMessageHeader::new();
+    header.ihr_disabled = true;
+    header.set_dst(masterchain_address(dst));
+    header.value = CurrencyCollection::with_grams(value);
+    Message::with_int_header(header)
+}
```
Comment (awnion) [inline tvm_executor/tests/common/mod.rs]: Make the destination workchain explicit so tests can cover wc0 instead of always routing to masterchain.

```suggestion
pub fn outbound_internal_message(dst: u8, value: u64, workchain_id: i8) -> Message {
    let mut header = InternalMessageHeader::new();
    header.ihr_disabled = true;
    header.set_dst(
        MsgAddressInt::with_standart(
            None,
            workchain_id as i32,
            UInt256::with_array([dst; 32]).into(),
        )
        .unwrap(),
    );
    header.value = CurrencyCollection::with_grams(value);
    Message::with_int_header(header)
}
```

**Review by awnion — COMMENTED** [review]
Additional applyable suggestions for the refined review.

**Review by melsomino — APPROVED** [review]
